src/main/java/org/launchcode/techjobs/oo/Job.java

```diff
@@ -1,25 +1,137 @@
+        return coreCompetency;
+    }
+
+    //setters
+    public void setName(String name) {
+        this.name = name;
+    }
+    public void setEmployer(Employer employer) {
+        this.employer = employer;
+    }
+    public void setLocation(Location location) {
+        this.location = location;
+    }
+    public void setPositionType(PositionType positionType) {
+        this.positionType = positionType;
+    }
+    public void setCoreCompetency(CoreCompetency coreCompetency) {
+        this.coreCompetency = coreCompetency;
+    }
+
+    //constructors
+    public Job(){
+        id = nextId;
+        nextId++;
+    }
+    public Job(String aName, Employer aEmployer, Location aLocation, PositionType aPositionType, CoreCompetency aCoreCompetency){
+        this();
+        name = aName;
+        employer = aEmployer;
+        location = aLocation;
+        positionType = aPositionType;
+        coreCompetency = aCoreCompetency;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Job job = (Job) o;
+        return id == job.id;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id);
+    }
+
+    @Override
+    public String toString() {
+        String value = "";
+        if((getName() == null || getName() == "") && getEmployer() == null && getLocation() == null && getPositionType() == null && getCoreCompetency() == null){
+            return "\nOOPS! This job does not seem to exist.\n";
+        }
+
+        try {
+            value =  String.format("\n" +
+                            "ID: %s\n" +
+                            "Name: %s\n" +
+                            "Employer: %s\n" +
+                            "Location: %s\n" +
+                            "Position Type: %s\n" +
+                            "Core Competency: %s\n",
+                    id == 0? DEFAULT_MESSAGE : id,
+                    getValueFromField("Name",true),
+                    getValueFromField("Employer",true),
+                    getValueFromField("Location",true),
+                    getValueFromField("PositionType",true),
+                    getValueFromField("CoreCompetency",true));
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
+        } catch (InvocationTargetException e) {
+            e.printStackTrace();
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
+        }
 
-    // TODO: Add custom equals and hashCode methods. Consider two Job objects "equal" when their id fields
-    //  match.
+        return value;
+    }
+    public String getValueFromField(String fieldName, Boolean substituteEmpty) throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, ClassNotFoundException {
```

Comment:
This is a very creative way of doing this. Great job! 

This is the way I saw it. Each one of the classes (Employer, Location, etc) extend the `JobField` class. Which means that they all have the `getValue()` method. I would suggest creating these overloaded methods instead:
```

    public String getValueFromField(JobField field) {
        if (field == null || field.getValue() == "") field.setValue(DEFAULT_MESSAGE);
        return field.getValue();
    }

    public String getValueFromField(String name) {
        if (name == null || name == "") name = DEFAULT_MESSAGE;
        return name;
    }
```

And so then, in your string on line 99-103, you can call the method by just passing the name, employer, location, etc. fields.